### ISSUE TITLE
eliminate setstate warnings

### ISSUE THREAD

**Issue by bermarte** [bermarte/Restful-chinook#31]
- eliminate setstate warnings

**Comment by bermarte** [comment]
# How to reproduce it:

run the back-end:
`npm run dev`
run the client:
`cd client`
`npm start`

Once the client opens, check the web console:
![warning](https://user-images.githubusercontent.com/979362/114391342-a9702c80-9b97-11eb-9217-5c0edddf6f1e.png)

This is something happening in all the components used (Albums.js, Genres.js, MediaTypes.js, Playlists.js and Traks.js).  
warning: `Do not mutate state directly. Use setState()`

- for example, in Albums.js, line 50:
```javascript
    handleInputChange(event){
        let val = event.target.value;
        console.log('text', val);
        //set value for editItem
        this.state.album = val;
    }
```
if I would use `this.setState({ "album" : val })` I wouldn't be able to edit the value of that field,  because `setState()`is async and the event (triggered by `onChange`) take places before `setState()` is completed.

- line 147:
```jsx
<input
type="text"
className="form-control"
id={album.AlbumId}
 defaultValue={album.Title}
onChange={(event) => this.handleInputChange(event)}
 />
```
![cli](https://user-images.githubusercontent.com/979362/114424279-3cb95a00-9bb8-11eb-96ea-59fc44a52c03.png)

**Comment by bermarte** [comment]
Fixed in a simple way with commit 31d3cdc53151de315a97820dac62e6b31ce6c89d
